### PR TITLE
Fix Javadoc view not starting

### PR DESCRIPTION
**Pull request by akurtakov**
Null values passed are preserved and sent to HTMLBuilder which sets them to default colors if needed.

### CODE REVIEW

**Comment by akurtakov** [comment]
The actual code fix  for https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/973

**Comment by akurtakov** [comment]
Not the brightest fix but better than copy this code everywhere fromRGB is used.

**Comment by akurtakov** [comment]
@iloveeclipse What's your opinion here?

File: bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/html/HTMLPrinter.java
Comment (iloveeclipse) [inline bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/html/HTMLPrinter.java:66]: The problem here is the second public constructor that initializes everything **without** null check. Means, this would not work for client code that creates `HTMLBuilder(RGB bg, RGB fg, RGB link, RGB alink)` with some "null" color.
I haven't reviewed original PR that introduced this constructor, and it should not be used yet widely. So I think this PR should add some assert in that constuctor.

Comment (akurtakov) [reply bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/html/HTMLPrinter.java:66]: The NPE seen happens before even getting into HTMLBuilder but I'll add guards for that too.

Comment (akurtakov) [reply bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/html/HTMLPrinter.java:66]: No null colors now with latest revision.

**Comment by akurtakov** [comment]
Once verifications are green I plan to merge this one unless there are objections as this is will fix the immediate issue. Further fixes/improvements can be done after that in other PRs.

File: bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/html/HTMLPrinter.java
Comment (iloveeclipse) [reply bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/html/HTMLPrinter.java:66]: Thanks

**Review by iloveeclipse — APPROVED** [review]

**Comment by akurtakov** [comment]
Unrelated test failure on MacOS only.

**Pull request merged**
